### PR TITLE
Add comprehensive test suite and documentation

### PR DESCRIPTION
**Pull request by ericmjl**
## Summary
- Add unit tests for Parser, shortcode handlers, error handling
- Add Lektor integration tests with demo site fixture
- Update README with modern installation and development instructions
- Test coverage for positional args, keyword args, block shortcodes

## Stack
This is **PR 4 of 4** in the modernization stack.
- **Depends on:** #14 (Type annotations) → #13 (CI/CD) → #12 (Build system)
- Please merge in order: #12, #13, #14, then this PR

## Testing
- All 11 tests pass
- Includes both unit tests and Lektor integration tests

### CODE REVIEW

**Comment by skorokithakis** [comment]
Hm, this PR alone is three times the size of the original code. I'm all for LLMs, but I'm going to have to review the code, and I can't go over thousands of lines to what's a 400 line plugin. If you'd like to submit a few minimal changes, that would be great, but this is a bit overwhelming for me to review, so PRs this big are unlikely to get merged any time soon :/.

**Comment by ericmjl** [comment]
@skorokithakis sorry about the misunderstanding here, but I've made the PRs in a "stacked" fashion, which should mean that if we review the earlier PRs first, with the changes merged in (via rebase), you'll get a clean commit history, and the size of the diff on the subsequent PRs should become smaller. Hopefully that helps?

**Comment by skorokithakis** [comment]
No I know, the changes I'm referring to are things like the pyproject.toml being so large, with multiple settings. I don't know whether ignoring those classes of errors is correct, or whether the code should change because it's actually wrong, and it's hard for me to test all of them. Can you please clarify why you made those choices?

**Comment by ericmjl** [comment]
Ah okay, yes, happy to do so. I'm going to leave comments on the individual PRs, rather than on this one, as it'll help make clear which changes really belong to their respective ones.

**Comment by ericmjl** [comment]
Alrighty, having left comments on the other three PRs, here's what this PR is focused on. Here, we add a comprehensive test suite to verify the plugin works correctly. 

Tests cover shortcode parsing, error handling, and Lektor integration. Importantly, it also includes a fixture demo site to test real-world scenarios. The README is also updated with testing instructions. 

Tests run via the CI from PR #13. Happy to discuss coverage priorities or simplify if this feels like too many tests.

File: tests/test_shortcodes.py
Comment (ericmjl) [inline tests/test_shortcodes.py:1]: These unit tests directly test the scodes.Parser module - things like registering handlers, parsing shortcodes, handling errors, and edge cases (escaping, nesting, etc.). They run fast (no Lektor dependencies) and caught several bugs during development. Each test is focused on one specific behavior to make failures easy to diagnose.

File: tests/test_lektor_integration.py
Comment (ericmjl) [inline tests/test_lektor_integration.py:1]: Integration tests that spin up a real Lektor project (using fixtures/demo-site/) and verify the plugin works end-to-end. These test the full plugin lifecycle: loading config, processing markdown, rendering templates. Slower than unit tests but catch issues that only appear when integrated with Lektor.

I decided to ask AI to make this test suite as I had trouble installing lektor-shortcodes on my personal website (https://ericmjl.github.io). Once I figured out how - it was a dependency chain issue - I thought it'd be a good idea to have an integration test with Lektor in the loop.

File: tests/fixtures/demo-site/Demo Site.lektorproject
Comment (ericmjl) [inline tests/fixtures/demo-site/Demo Site.lektorproject:1]: This file, and everything under the tests/fixtures/demo-site, are intended to be part of the lektor integration test.

**Review by skorokithakis — APPROVED** [review]

**Comment by skorokithakis** [comment]
Can you please fix pre-commit so we can merge?

**Comment by ericmjl** [comment]
Coming up!

**Comment by ericmjl** [comment]
And we're done! @skorokithakis thank you for doing the review, really appreciate it!

File: README.md [hunk 1]
```diff
@@ -91,6 +109,61 @@ request a different section, and it will always include shortcodes in `global` n
 matter what).
 
 
+Development
```
Comment (skorokithakis) [inline README.md]: This bit is standard LLM verbosity that just makes the README long, we shouldn't have instructions on how to install uv here. Can you please remove this whole section?

Comment (ericmjl) [reply README.md]: Done!

**Review by skorokithakis — APPROVED** [review]

**Comment by skorokithakis** [comment]
Thank you!
